### DISCUSSION TITLE
Is there a way to automatically configure service's parent?

### DISCUSSION THREAD

**Discussion by steevanb** [symfony/symfony#51505] [Q&A] [answered]
Hi :)

I have this classes:
```php
abstract class AbstractFoo
{
     public function setFoo(string $foo): static {...}
}

class Foo extends AbstractFoo
{}
```

I define AbstractFoo to be an abstract service and add a call on it:
```php
$this
    ->servicesConfigurator
    ->set(AbstractFoo::class)
    ->abstract()
    ->call('setFoo', ['foo']);
```

But i need to configure Foo too, to add parent(), it's not automatic:
```php
$this
    ->servicesConfigurator
    ->get(Foo::class)
    ->parent(AbstractFoo::class);
```

Is there a way to not manually indicate the parent, it's obvious because Foo extends AbstractFoo?

**Comment by mbabker** [comment, marked as answer]
It's not something that's automatically done right now.  Just because a class extends another doesn't necessarily mean that there is a corresponding parent service definition, so checking the class hierarchy to automatically set a parent service might end up being one of those things where the performance hit might not be worth the extra code.

**Reply by steevanb** [reply]
I was afraid of this answer, for the reasons you mention ;)

**Comment by chapterjason** [comment]
You could use the decorator pattern and a compiler pass to set the actual decorator class.

- https://symfony.com/doc/current/service_container/service_decoration.html
- https://symfony.com/doc/current/service_container/compiler_passes.html
